### PR TITLE
DYN-6397 Fix crash when parent node is deleted while node autocomplete menu is displayed

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

https://jira.autodesk.com/browse/DYN-6397

When a user deletes a node that has the node autocomplete menu open, it leaves the menu dissociated with the node and when a user clicks on any item on the menu, Dynamo crashed.

This PR fixes this case by closing the associated node autocomplete menu when the parent node is deleted.
On delete key press event any open node autocomplete menu will be closed.

DynamoSandbox_CuQpqGJ8S5

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Fix crash when parent node is deleted while node autocomplete menu is displayed

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -2785,13 +2785,22 @@ internal void Delete(object parameters)
 
             var command = new DynamoModel.DeleteModelCommand(Guid.Empty);
             this.ExecuteCommand(command);
+            OnNodeDeleted(this, EventArgs.Empty);
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:2788]: This command seems to be a general entrance for not only nodes but all graph elements, I would suggest trigger event on a different place to be more accurate, e.g. RemoveAndDisposeNode() at WorkspaceModel.cs

**Review by QilongTang — COMMENTED** [review]
Good approach! One comment

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs (hunk 1 above)
Comment (zeusongit) [reply src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: Great tip, now only when the targeted parent node is deleted the node autocomplete menu gets dismissed.

File: src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs [hunk 2]
```diff
@@ -530,6 +532,7 @@ internal void PopulateAutoCompleteCandidates()
         {
             if (PortViewModel == null) return;
 
+            dynamoViewModel.CurrentSpaceViewModel.Model.NodeRemoved += NodeViewModel_Removed;
```
[the remark this answers is not in the document]
Comment (QilongTang) [reply src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs:535]: One way is to pop the event all the way to DynamoViewModel level?

**Comment by zeusongit** [comment]
Added a test, and reviewed the comments

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
Are there any regessions?

**Comment by zeusongit** [comment]
> Are there any regressions?

I see 18 test failed, but seems un-related.

**Comment by QilongTang** [comment]
Maybe try the self serve on master-15? Even after I update your branch with master, there are still a bunch failures.

**Comment by zeusongit** [comment]
> Maybe try the self serve on master-15? Even after I update your branch with master, there are still a bunch failures.

Tried [master-15 ](https://master-15.jenkins.autodesk.com/view/DYN/job/DYN-DevCI_Self_Service/1347/), but each run gets me a different result, not sure what the problem is, I will wait for the ongoing issue with async test runs to resolve.

File: src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs [hunk 3]
```diff
@@ -351,12 +365,13 @@ private void OnMoreInfoClicked(object sender, RoutedEventArgs e)
...
         internal void CloseAutocompletionWindow(object sender, RoutedEventArgs e)
         {
-            OnRequestShowNodeAutoCompleteSearch(ShowHideFlags.Hide);
+            CloseAutoCompletion();
         }
 
         internal void CloseAutoCompletion()
         {
             OnRequestShowNodeAutoCompleteSearch(ShowHideFlags.Hide);
+            ViewModel.OnNodeAutoCompleteWindowClosed();
```
Comment (pinzart90) [inline src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs:374]: Looks like the ViewModel could be null here
https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/12457/testReport/Dynamo.Tests/WorkspaceOpeningTests/ClearingWorkspaceResetsPositionAndZoom/

**Comment by zeusongit** [comment]
No regression, all checks passed, merging.

**Pull request merged**